### PR TITLE
Vendor gardener/gardener@v1.36.0

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind enhancement
/platform aws

Fixes https://github.com/gardener/gardener-extension-provider-aws/issues/430

Follow-ups that can be performed separately:
- [ ] Do not rely on the deprecated `hack/install-requirements.sh` and adapt to the new way of installing tool dependencies (ref https://github.com/gardener/gardener/pull/4879).

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/apis/aws/validation/infrastructure.go [hunk 1]
```diff
@@ -153,12 +153,13 @@ func ValidateInfrastructureConfig(infra *apisaws.InfrastructureConfig, nodesCIDR
...
-		allErrs = append(allErrs, vpcCIDR.ValidateNotSubset(pods, services)...)
+		allErrs = append(allErrs, vpcCIDR.ValidateNotOverlap(pods, services)...)
 	}
 
 	// make sure that VPC cidrs don't overlap with each other
-	allErrs = append(allErrs, cidrvalidation.ValidateCIDROverlap(cidrs, cidrs, false)...)
-	allErrs = append(allErrs, cidrvalidation.ValidateCIDROverlap([]cidrvalidation.CIDR{pods, services}, cidrs, false)...)
+	allErrs = append(allErrs, cidrvalidation.ValidateCIDROverlap(cidrs, false)...)
+	allErrs = append(allErrs, pods.ValidateNotOverlap(cidrs...)...)
+	allErrs = append(allErrs, services.ValidateNotOverlap(cidrs...)...)
```
Comment (ialidzhikov) [inline pkg/apis/aws/validation/infrastructure.go:162]: @kon-angelo can you double-check whether this change is fine as adaptation for the breaking changes in https://github.com/gardener/gardener/pull/4829? Thanks in advance.

Comment (kon-angelo) [reply pkg/apis/aws/validation/infrastructure.go:162]: looks good

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Pull request merged**